src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -550,15 +548,19 @@ class MavlinkStreamSysStatus : public MavlinkStream
 
 	bool send(const hrt_abstime t)
 	{
-		vehicle_status_s status = {};
-		cpuload_s cpuload = {};
-		battery_status_s battery_status = {};
-
+		vehicle_status_s status;
 		const bool updated_status = _status_sub->update(&_status_timestamp, &status);
-		const bool updated_cpuload = _cpuload_sub->update(&_cpuload_timestamp, &cpuload);
-		const bool updated_battery = _battery_status_sub->update(&_battery_status_timestamp, &battery_status);
 
-		if (updated_status || updated_battery || updated_cpuload) {
+		if (updated_status) {
+
+			// if vehicle_status updated always grab latest cpuload and battery_status
+
+			battery_status_s battery_status = {};
+			_battery_status_sub->update(&battery_status);
```

Comment:
Do you know you will always have valid data for all messages because both `battery_status` and `cpuload` are published more frequently than `vehicle_status`?

If so, aren't you at risk of using old/queued data from `battery_status` and `cpuload`?

Reply:
In any normal usage we can expect to have them continuously (if we ever had them). Deciding if it's worth handling the failure case is a fair point.

For example if you had a smart battery and lost communication. The now separate BATTERY_STATUS message would stop, but SYS_STATUS battery fields would be stale.

Reply:
What I meant was when the system is running as normal and e.g. `battery_status` publishes at 3x the rate of `vehicle_status`, the messages would stack up and you'd use an old message.

I don't know how big the stack is but I have seen something to the effect of the following in other places:

```
while(_battery_status_sub->update(&battery_status)) {}
```

Reply:
That's only an issue for topics that are queued. Currently that's vehicle_command (+ack), subsystem_info, and transponder_report.

Reply:
Ah, good to know. Thanks!